packages/mcp/src/microsoft/teams/mcp/ai_plugin.py

```diff
@@ -0,0 +1,262 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+import asyncio
+import json
+import logging
+import time
+from typing import Any, Awaitable, Callable, Dict, List, Optional, Tuple, Union
+
+from microsoft.teams.ai.function import Function
+from microsoft.teams.ai.plugin import BaseAIPlugin
+from microsoft.teams.common.logging import ConsoleLogger
+from pydantic import BaseModel
+
+from mcp import ClientSession
+from mcp.types import TextContent
+
+from .transport import create_transport
+
+REFETCH_TIMEOUT_MS = 24 * 60 * 60 * 1000  # 1 day
+
+
+class McpToolDetails(BaseModel):
```

Comment:
nit: could we have this in a separate file - same w cached value? separation of concerns

Reply:
Moved em to their own model files